packages/fether-react/src/Accounts/CreateAccount/AccountImportOptions/AccountImportOptions.js

```diff
@@ -88,41 +108,64 @@ class AccountImportOptions extends Component {
     }
   };
 
+  /**
+   * The `chainId$` and `chainName$` from light.js corresponds to `chainID` in the
+   * Genesis configs contained in: paritytech/parity-ethereum/ethcore/res/ethereum
+   * and was introduced in EIP-155 https://github.com/ethereum/EIPs/blob/master/EIPS/eip-155.md
+   * to prevent replay attacks between `foundation` and `classic` chains, which both have
+   * `networkID` of `1`.
+   */
   handleSignerImported = async ({ address, chainId: chainIdString }) => {
     const {
+      chainId: currentChainIdBN,
+      chainName,
       createAccountStore: { importFromSigner }
     } = this.props;
 
     if (!address || !chainIdString) {
       this.setState({ error: 'Invalid QR code.' });
+
       return;
     }
 
-    const chainId = parseInt(chainIdString);
+    const signerChainId = parseInt(chainIdString);
+
+    if (!BigNumber(signerChainId).eq(currentChainIdBN)) {
```

Comment:
`currentChainIdBN.eq(signerChainId)` is easier to read, and you don't even need to parseInt

Reply:
i've addressed this in commit 1926ab740af8da3bf7e07e9ad8ed3064b317dedb